File: app/src/main/java/com/lucciola/haskellanywhereforandroid/uiaction/UIActionFactory.java
package com.lucciola.haskellanywhereforandroid.uiaction;

import com.lucciola.haskellanywhereforandroid.Action;
import com.lucciola.haskellanywhereforandroid.View;

/**
 * Created by hiden on 2017/06/19.
 */

public class UIActionFactory {

    public static UIAction createUIAction(Action action, View view) {
        UIAction result = null;
        switch (action.getMode()) {
            case Action.MODE_HASKELL: {
                result = new HaskellAction(action.getProgramManager(), view);
                break;
            }
            case Action.MODE_FUNCTION: {
                result = new FunctionAction(action.getProgramManager(), view);
                break;
            }
            case Action.MODE_COMMAND: {
                result = new CommandAction(action.getProgramManager(), view);
                break;
            }
        }
        return result;
    }
}

File: app/src/main/kotlin/com/lucciola/haskellanywhereforandroidkt/uiaction/UIActionFactory.kt
package com.lucciola.haskellanywhereforandroidkt.uiaction

import com.lucciola.haskellanywhereforandroidkt.Action
import com.lucciola.haskellanywhereforandroidkt.View

/**
 * Created by hiden on 2017/06/19.
 */

object UIActionFactory {

    fun createUIAction(action: Action, view: View): UIAction? {
        var result: UIAction? = null
        when (action.mode) {
            Action.MODE_HASKELL -> {
                result = HaskellAction(action.programManager, view)
            }
            Action.MODE_FUNCTION -> {
                result = FunctionAction(action.programManager, view)
            }
            Action.MODE_COMMAND -> {
                result = CommandAction(action.programManager, view)
            }
        }
        return result
    }
}
